rasa/shared/core/domain.py

```diff
@@ -309,8 +311,93 @@ def merge_lists_of_dicts(
 
         return self.__class__.from_dict(combined)
 
+    @staticmethod
+    def merge_domain_dicts(
+        domain1: Dict, domain2: Dict, override: bool = False
+    ) -> Dict[Text, Any]:
+        """Merge this domain dict with another one, combining their attributes.
+
+        Used when multiple domain files are configured (in a single directory).
+
+        List attributes like ``intents`` and ``actions`` will be deduped
+        and merged. Single attributes will be taken from `domain1` unless
+        override is `True`, in which case they are taken from `domain2`.
+
+        This is a fix for #10137
+        """
+        if not domain2:  # or domain2.is_empty():
+            return domain1
+
+        if domain1 == {}:  # .is_empty():
+            return domain2
+
+        domain_dict = domain2  # .as_dict(mergeflag=True)
+        combined = domain1  # .as_dict(mergeflag=True)
```

Comment:
```suggestion
        if not domain2:
            return domain1

        if domain1 == {}:
            return domain2

        domain_dict = domain2
        combined = domain1
```

Reply:
I think these were left here by accident? Maybe you copied the old merge function to edit it and forgot to delete them?

Reply:
@aeshky 
> Does this need fixing on main too?
We're going to confine this fix to 2.8 for now as we are considering an overhaul to the Domain class.